testFlight1/lib/teensyShared/rocket.cpp

```diff
@@ -0,0 +1,244 @@
+ 
+                // If the acceleration lasts long enough, drogue is detected
+                if (rocketTimers.drogue_timer > drogue_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_DROGUE;
+                }
+            break;
+ 
+            case STATE_DROGUE:
+                //If the rocket has higher acceleration
+                //!locking mutex to get data from sensorData struct
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > main_thresh_min) {
+                    rocketTimers.main_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_MAIN_DETECT;
+                }
+            break;
+ 
+            case STATE_MAIN_DETECT:
+                //If the acceleration was too brief, go back to drogue
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < main_thresh_min
+                        || pointer_struct->sensorDataPointer->state_data.state_vz > main_thresh_max) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE;
+                    break;
+                }
+ 
+                // measure the length of the main time
+                rocketTimers.main_timer =
+                    chVTGetSystemTime() - rocketTimers.main_time;
+ 
+                // If the low velocity lasts long enough, main is detected
+                if (rocketTimers.main_timer > main_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_MAIN;
+                }
+            break;
+ 
+            case STATE_MAIN:
+                //If the rocket has 0 velocity
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > landed_thresh) {
+                    rocketTimers.landing_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_LANDED_DETECT;
+                }
+            break;
+ 
+            case STATE_LANDED_DETECT:
+                //If the 0 velocity was too brief, go back to main
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < landed_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE;
+                    break;
+                }
+ 
+                // measure the length of the landing
+                rocketTimers.landing_timer =
+                    chVTGetSystemTime() - rocketTimers.landing_time;
+ 
+                // If the 0 velocity lasts long enough, landed is detected
+                if (rocketTimers.landing_timer > landing_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_LANDED;
+                }
+            break;
+ 
+            case STATE_LANDED:
+                // TODO
```

Comment:
Nit: Change the comment here to say this is the final state. Or maybe add some kind of pass statement?